examples/with-redux-observable/README.md

```diff
@@ -29,24 +29,56 @@ npm run dev
 
 
 ### The idea behind the example
-Example is a page that renders information about Star-Wars characters. It fetches new character  
-every 3 seconds having the initial character fetched on a server.
+
+This example is a page that renders information about Star-Wars characters. It
+fetches new character every 3 seconds having the initial character fetched on
+a server.
 
 Example also uses `redux-logger` to log every action.
 
 ![demo page](demo.png)
 
-The main problem with integrating Redux, Redux-Observable and Next.js is probably making initial requests
-on a server. That's because it's not possible to wait until epics are resolved in `getInitialProps` hook. 
+The main problem with integrating Redux, Redux-Observable and Next.js is
+probably making initial requests on a server. That's because, the
+`getInitialProps` hook runs on the server-side before epics have been made available by just dispatching actions.
+
+However, we can access and execute epics directly. In order to do so, we need to
+pass them an Observable of an action and they will return an Observable:
+
+```js
+static async getInitialProps({ store, isServer }) {
+  const resultAction = await rootEpic(
+    of(actions.fetchCharacter(isServer)),
+    store
+  ).toPromise(); // we need to convert Observable to Promise
+  store.dispatch(resultAction)};
+```
+
+Note: we are not using `AjaxObservable` from the `rxjs` library; as of rxjs
+v5.5.6, it will not work on both the server- and client-side. Instead we call
+the default export from
+[universal-rx-request](https://www.npmjs.com/package/universal-rx-request) (as
+`ajax`).
+
+We transform the Observable we get from `ajax` into a Promise in order to await
+its resolution. That resolution should be a action (since the epic returns
+Observables of actions). We immediately dispatch that action to the store. 
+
+This server-side solution allows compatibility with Next. It may not be
+something you wish to emulate. In other situations, calling or awaiting epics
+directly and passing their result to the store would be an anti-pattern. You
+should only trigger epics by dispatching actions. This solution may not
+generalise to resolving more complicated sets of actions. 
```

Comment:
I actually kind of like it a lot, it still forces you to be transactional. The real risk (?) is that you'll impact startup time by invoking everything in your root epic. I think it's good to share code amongst the backend and frontend to have a consistent state and a performant response for users.